### PR TITLE
Only fight sandworms when it makes sense.

### PR DESCRIPTION
**Pull request by phulin**
Also fixes the bug where we don't equip champagne.

### CODE REVIEW

File: src/outfit.ts [hunk 1]
```diff
@@ -85,11 +87,8 @@ export function freeFightOutfit(requirements: Requirement[] = []): void {
   );
 
   const finalRequirement = new Requirement(parameters, {
-    forceEquip: forceEquip,
-    preventEquip: [
-      ...preventEquip,
-      ...$items`broken champagne bottle, Spooky Putty snake, Spooky Putty mitre, Spooky Putty leotard, Spooky Putty ball, papier-mitre, smoke ball`,
-    ],
+    forceEquip,
+    ...(preventEquip.length > 0 ? { preventEquip } : {}),
```
Comment (horrible-little-slime) [inline src/outfit.ts:91]: We use preventEquip for the bjorn/crown much of the time--wouldn't this perpetually override our default settings?

Comment (phulin) [reply src/outfit.ts:91]: oh okay, I'll have to scrub the code for that

File: src/fights.ts [hunk 2]
```diff
@@ -348,10 +347,11 @@ export function dailyFights(): void {
...
         const startLectures = get("_pocketProfessorLectures");
         useFamiliar($familiar`Pocket Professor`);
-        meatOutfit(true, [
-          ...fightSource.requirements,
-          new Requirement([], { forceEquip: $items`Pocket Professor memory chip` }),
-        ]);
+        const requirement = Requirement.merge(fightSource.requirements);
+        const forceEquip = requirement.maximizeOptions.forceEquip ?? [];
+        forceEquip.push($item`Pocket Professor memory chip`);
+        requirement.maximizeOptions.forceEquip = forceEquip;
```
Comment (horrible-little-slime) [inline src/fights.ts:353]: Does this work? `maximizeOptions` is a getter function, not the property itself.

File: src/outfit.ts [hunk 3]
```diff
@@ -58,15 +57,12 @@ const bestAdventuresFromPants =
...
-  const compiledRequirements = requirement ?? new Requirement([], {});
-  const compiledOptions = compiledRequirements.maximizeOptions;
-  const compiledParameters = compiledRequirements.maximizeParameters;
 
-  const forceEquip = compiledOptions.forceEquip ?? [];
-  const bonusEquip = compiledOptions.bonusEquip ?? new Map<Item, number>();
-  const preventEquip = compiledOptions.preventEquip ?? [];
-  const preventSlot = compiledOptions.preventSlot ?? [];
-  const parameters = compiledParameters;
+  const parameters = requirement?.maximizeParameters ?? [];
```
Comment (horrible-little-slime) [inline src/outfit.ts:61]: We should make this a function that we add to libram; something that essentially explodes a Requirement object into a bunch of consts.

**Review by horrible-little-slime — COMMENTED** [review]
It feels like the solution here might be a rehaul of the Requirement class in libram, or an expansion of it.

File: src/fights.ts [hunk 4]
```diff
@@ -1474,6 +1479,13 @@ const freeRunFightSources = [
   ),
 ];
 
+function sandwormRequirement() {
+  return new Requirement(
+    ["100 Item Drop"],
+    get("garbageChampagneCharge") > 0 ? { forceEquip: $items`broken champagne bottle` } : {}
```
Comment (horrible-little-slime) [inline src/fights.ts:1485]: does anything verify that you own the champagne bottle?

Comment (phulin) [reply src/fights.ts]: done

File: src/outfit.ts (hunk 3 above)
Comment (phulin) [reply src/outfit.ts:41]: after more thought i think maybe we should do something like your requirement builder thing

File: src/fights.ts (hunk 2 above)
Comment (phulin) [reply src/fights.ts:343]: yeah

**Comment by phulin** [comment]
I don't think this is the cleanest solution at this point but it's an improvement. I suggest we merge and continue to think about improving `Requirement`. Thoughts?

**Pull request merged**